apps/browser/app/api/goal/generate-title/route.ts

```diff
@@ -0,0 +1,73 @@
+import { getAuthOptions } from "@/lib/api/authOptions";
```

Comment:
not sure this needs to be apart of the `goal/` api. Might make more sense to add under the `chat/` api (`chat/generate-title`)

Reply:
Or maybe `get-title` as it may just return the existing title.

Reply:
i think `generate-title` makes the more sense, as it never returns the existing title (if it exists)